src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -1,19 +1,150 @@
+    private final double loadFactor = 0.75;
+    private int treshold = (int)(capacity * loadFactor);
+    private Node[] table = new Node[capacity];
 
     @Override
     public void put(K key, V value) {
-
+        if (size + 1 > treshold) {
+            resize();
+        }
+        for (Node element : table) {
+            if (element != null && (element.getValue().equals(value)
+                    || element.getValue() == value)) {
+                return;
+            }
+            if (element != null && element.getNext() != null) {
+                while (element != null) {
+                    if (element.getValue() == value || element.getValue().equals(value)) {
+                        return;
+                    }
+                    element = element.getNext();
+                }
+            }
+        }
+        if (key == null || indexForElement(hash(key),capacity) == 0) {
+            addToFirstPosition(key,value);
+            return;
+        }
+        if (table[indexForElement(hash(key),capacity)] != null
+                && table[indexForElement(hash(key),capacity)].getNext() == null) {
+            Node element = table[indexForElement(hash(key),capacity)];
+            if (element.getKey() == key) {
+                element.setValue(value);
+                return;
+            }
+            table[indexForElement(hash(key),capacity)]
+                    .setNext(new Node<>(hash(key),key,value,null));
+            size++;
+            return;
+        }
+        if (table[indexForElement(hash(key),capacity)] != null
+                && table[indexForElement(hash(key),capacity)].getNext() != null) {
+            Node element = table[indexForElement(hash(key),capacity)];
+            while (element.getNext() != null) {
+                if (element.getKey() == key) {
+                    element.setValue(value);
+                    return;
+                }
+                element = element.getNext();
+            }
+            element.setNext(new Node<>(hash(key),key,value,null));
+            size++;
+            return;
+        }
+        table[indexForElement(hash(key),capacity)] = new Node<>(hash(key),key,value,null);
+        size++;
     }
 
     @Override
     public V getValue(K key) {
+        Node element = table[indexForElement(hash(key),capacity)];
+        if (element == null) {
+            return null;
+        }
+        if (element.getNext() == null) {
+            return (V) element.getValue();
+        }
+        if (element.getNext() != null) {
+            while (element.getNext() != null) {
+                if (element.getKey() != null && element.getKey().equals(key)) {
+                    return (V) element.getValue();
+                }
+                if (element.getKey() == null && key == null) {
+                    return (V) element.getValue();
+                }
+                element = element.getNext();
+            }
+            return (V) element.getValue();
+        }
```

Comment:
```suggestion
           int index = getIndexFromHash(key);
        Node<K, V> returnNode = table[index];
        while (returnNode != null) {
            if (Objects.equals(returnNode.key, key)) {
                return returnNode.value;
            }
            returnNode = returnNode.nextNode;
        }
        return null;
        }
```

we can simplify this